Sources/Layout/AxisDependentLayout.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  AxisDependentLayout.swift
+//  CollectionKit
+//
+//  Created by Luke Zhao on 2017-08-24.
+//  Copyright © 2017 lkzhao. All rights reserved.
+//
+
+import UIKit
+
+public enum Axis{
+  case vertical, horizontal
+}
+
+public class AxisDependentLayout<Data>: CollectionLayout<Data> {
```

Comment:
I thought about protocol. but then every class have to store its own axis variable and also config their visibleIndexSorter according to their axis.